### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by medcelerate**

### CODE REVIEW

**Comment by michael-kotliar** [comment]
I would rather use `hints` instead of `requirements` for describing Docker containers. Otherwise, we **must** use containers.
https://www.commonwl.org/v1.1/CommandLineTool.html#Requirements_and_hints

**Comment by medcelerate** [comment]
How are hints set instead of requirements, is there a check in the cwl spec
if the commandline tool is available?

**Comment by michael-kotliar** [comment]
I think, it mostly depends on the specific workflow runner. I didn't test if cwltool checks whether the program is available without container or not, but this is what CWL specification says about the difference between hints and requirements:
>If an implementation cannot satisfy all requirements, or a requirement is listed which is not recognized by the implementation, it is a fatal error and the implementation must not attempt to run the process, unless overridden at user option.
>A hint is similar to a requirement; however, it is not an error if an implementation cannot satisfy all hints. The implementation may report a warning if a hint cannot be satisfied.

I usually use something like this:
```
hints:
- class: DockerRequirement
  dockerPull: biowardrobe2/manorm:v0.0.2
```

File: README.md [hunk 1]
```diff
@@ -11,16 +11,77 @@ The underlying tools are under one or more Free and Open Source Software license
...
-1. `*.cwl` files must be marked executable (`chmod a+x *.cwl`)
-1. All tool descriptions must have a software container. Use a container from biocontainers.pro if available
-1. If you use schema.org annotations, specify the schema using the RDF version: `$schemas: [ http://schema.org/version/latest/schema.rdf ]`
-1. Must be free from warning when `cwltool --validate` is run
-1. Tool descriptions should be motivated by a real world use of this tool in a workflow.
-   The description should focus on a single way of using the tool.
-   Signs that a tool description is including too much: lots of javascript; complicated data structures; every single flag is listed.
+## Naming Tools 📛
+
+Tools should follow the convention of being prefixed by the parent tool name and Camelcase like so i.e.
```
Comment (mr-c) [inline README.md:16]: usually I prefer lowercase for faster typing. Thoughts from other @common-workflow-library/bio-team people?

File: README.md [hunk 2]
```diff
@@ -11,16 +11,77 @@ The underlying tools are under one or more Free and Open Source Software license
...
+`
+BWA-Mem.cwl`
+or `
+BWA-Index.cwl
+`
+
+## Tool Feature Requirements 🆕
+
+In order to keep consistent with the biotools spec, the first 3 lines of tool wrappers should be as follows. Our CI/CD system checks for these so make sure to include them so they can be merged into the repo.
```
Comment (mr-c) [inline README.md]:
```suggestion
The first 3 lines of tool wrappers should be as follows. Our CI/CD system checks for these so make sure to include them so they can be merged into the repo.
```

File: README.md [hunk 3]
```diff
@@ -11,16 +11,77 @@ The underlying tools are under one or more Free and Open Source Software license
...
+#!/usr/bin/env cwl-runner
+cwlVersion: v1.0
+class: CommandLineTool
+```
+The first line allows the tool to be run as a single command.
+The second specifies the cwl version.
+<br/>
+
+**IMPORTANT!** 
+The current spec is at v1.1, the IIDSGT group commits tools at **v1.0** due to a limit in our job management system. It is recommended to use the current spec whenever possible.
```
Comment (mr-c) [inline README.md]:
```suggestion
We are using `cwlVersion` `v1.0` unless a `v1.1` feature is needed.
```

**Review by mr-c — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (medcelerate) [reply README.md:16]: We used camelcase so we can add some more descriptive naming from the description in case of a long form parameter.

Comment (michael-kotliar) [reply README.md:16]: I usually use lowercase letters and underscore if I want to separate parts of the name, but it mostly defined by a habit and non-case-sensitive hard-drive.

**Comment by medcelerate** [comment]
So maybe this is something we can keep as part of our internal style guide
and have it be something open for people’s coding habits?

File: README.md (hunk 1 above)
Comment (medcelerate) [reply README.md:16]: So maybe this is something we can keep as part of our internal style guide and have it be something open for people’s coding habits?

Comment (mr-c) [reply README.md:16]: Yeah, I think this may be premature optimization right now

Comment (michael-kotliar) [reply README.md:16]: @mr-c Should we merge this PR or there is something left to fix?

Comment (medcelerate) [reply README.md:16]: @mr-c I believe this is good.

**Pull request merged**